Smart-Table.debug.js

```diff
@@ -608,23 +619,34 @@
                 var index = scope.dataCollection.indexOf(dataRow);
                 if (index !== -1) {
                     selectDataRow(scope.dataCollection, scope.selectionMode, index, dataRow.isSelected !== true);
+                    scope.$emit('selectionChange', {item: dataRow});
                 }
             };
 
             /**
-             * select/unselect all the currently displayed rows
+             * select/unselect the rows within the given range
+             * @param start starting index of range
+             * @param end ending index of range (inclusive)
              * @param value if true select, else unselect
              */
-            this.toggleSelectionAll = function (value) {
-                var i = 0,
-                    l = scope.displayedCollection.length;
-
-                if (scope.selectionMode !== 'multiple') {
+            this.toggleSelectionMultiple = function (start, end, value) {
+                if (!isModeMultiple()) {
```

Comment:
Created this method (which is called by toggleSelectionAll) to specify a range that I want selected so that you can select multiple at a time.